lib/logstash/filters/bytes.rb

```diff
@@ -0,0 +1,116 @@
+# encoding: utf-8
+require "logstash/filters/base"
+require "logstash/namespace"
+
+# This filter will parse a given string as a computer storage
+# value (e.g. "123 MB" or "5.3GB") and add a new numeric field with
+# size in bytes
+#
+class LogStash::Filters::Bytes < LogStash::Filters::Base
+
+  PREFIX_POWERS = {
+    'k' => 1, # 1 kilobyte = 1024 ^ 1 bytes
+    'm' => 2, # 1 megabyte = 1024 ^ 2 bytes
+    'g' => 3, # 1 gigabyte = 1024 ^ 3 bytes
+    't' => 4, # 1 terabyte = 1024 ^ 4 bytes
+    'p' => 5, # 1 petabyte = 1024 ^ 5 bytes
+    'e' => 6, # 1 exabyte = 1024 ^ 6 bytes
+  }.freeze
+
+  DIGIT_GROUP_SEPARATORS = " _,."
```

Comment:
Is there potentially a case for making this configurable too (not necessarily in this initial commit)? Digits are commonly separated by spaces in Europe, and by an apostrophe in Switzerland.

Reply:
++ to making it configurable. I'd also like to defer that to a follow up PR as this one's pretty large already.